atlasdb-api/src/main/java/com/palantir/atlasdb/keyvalue/api/Cell.java

```diff
@@ -62,10 +67,18 @@ public static boolean isNameValid(byte[] name) {
         return name != null && name.length > 0 && name.length <= MAX_NAME_LENGTH;
     }
 
-    public static void validateNameValid(byte[] name) {
+    private void validateNameValid(byte[] name) {
         Preconditions.checkNotNull(name, "name cannot be null");
         Preconditions.checkArgument(name.length > 0, "name must be non-empty");
-        Preconditions.checkArgument(name.length <= MAX_NAME_LENGTH, "name must be no larger than " + MAX_NAME_LENGTH);
+
+        String lengthErrorMessage = "name must be no longer than " + MAX_NAME_LENGTH;
+        if (log.isDebugEnabled()) {
+            lengthErrorMessage += ". Cell creation that was attempted was: " + this
+                    + "; since the vast majority of people encountering this problem are using unbounded Strings as"
+                    + " components, it may aid your debugging to know the ASCII interpretation of the bad field was:"
+                    + " [" + new String(name, StandardCharsets.US_ASCII) + "]";
```

Comment:
@clockfort said: "The weirdness here is because I don't want to put actual data in the logs normally. Happy to take suggestions/patches for a cleaner way to do it."

So this should return something like `48656c6c6f` if the actual stored value is `Hello`?

You should use `UTF_8` instead of `US_ASCII`.